polkadot/xcm/xcm-builder/src/universal_exports.rs

```diff
@@ -710,3 +713,29 @@ mod tests {
 		}
 	}
 }
+
+/// An adapter for the implementation of `ExporterFor`, which attempts to find the
+/// `(bridge_location, payment)` for the requested `network` and `remote_location` and `xcm`
+/// in the provided `T` table containing various exporters.
+pub struct NetworkWithXcmExportTable<T, M>(core::marker::PhantomData<(T, M)>);
+impl<T: Get<Vec<NetworkExportTableItem>>, M: Contains<Xcm<()>>> ExporterFor
```

Comment:
I’m not convinced we need both `NetworkExportTable` and `NetworkWithXcmExportTable`. Why not just modify the existing `NetworkExportTable` to include this `M` / `M::contains` logic? Wherever `NetworkExportTable` is used, we can simply add `NetworkExportTable<.., Everything>`.

Reply:
It's a low level primitive and the change will impact current codes/tests a lot, I'd prefer to minimize the change required.

Reply:
> It's a low level primitive and the change will impact current codes/tests a lot, I'd prefer to minimize the change required.

I assume that at some point you will remove support for `v1`, correct? If so, then `NetworkWithXcmExportTable` will no longer be needed. Therefore, I suggest moving `NetworkWithXcmExportTable` to the `bridges/snowbridge` submodule for now.

If it needs to stay here, I would suggest modifying `NetworkExportTable`, yes, it is a breaking change (just like XCMv5, which this feature is waiting for, right?), but it is clearer than having both `NetworkWithXcmExportTable` and `NetworkExportTable`.

Reply:
Moving `NetworkWithXcmExportTable` to the bridges/snowbridge submodule with https://github.com/paritytech/polkadot-sdk/pull/6074/commits/44dbb6caa8ee81be1e61880975355ba40d58017d